document not correct for `qm.alive`

```python
    # Ensures a clean and graceful shutdown of the job processing system.
    def handle_signal(*_args) -> None:
        qm.alive = False
```

and mypy:

```text
q\worker.py:230: error: Incompatible types in assignment (expression has type "bool", variable has type "Event")  [assignment]
```

should it be `qm.alive.set()`? and why set `alive` event to mark `QueueManager` as "dead"?

--- a/README.md
+++ b/README.md
@@ -4,8 +4,8 @@
 
 ---
 
-📚 **Documentation**: [Explore the Docs](https://pgqueuer.readthedocs.io/en/latest/)  
-🔍 **Source Code**: [View on GitHub](https://github.com/janbjorge/pgqueuer/)  
+📚 **Documentation**: [Explore the Docs](https://pgqueuer.readthedocs.io/en/latest/)
+🔍 **Source Code**: [View on GitHub](https://github.com/janbjorge/pgqueuer/)
 💬 **Join the Discussion**: [Discord Community](https://discord.gg/C7YMBzcRMQ)
 
 ---
@@ -122,4 +122,3 @@ PGQueuer is MIT licensed. See [LICENSE](LICENSE) for more information.
 
 ---
 Ready to supercharge your workflows? Install PGQueuer today and take your job management to the next level!
-

--- a/docs/queuemanager.md
+++ b/docs/queuemanager.md
@@ -34,7 +34,7 @@ async def main() -> None:
 
     # Ensures a clean and graceful shutdown of the job processing system.
     def handle_signal(signum: object, frame: object) -> None:
-        qm.alive = False
+        qm.shutdown.set()
 
     signal.signal(signal.SIGINT, handle_signal)
 
@@ -143,11 +143,11 @@ The QueueManager offers several configurable parameters to optimize job processi
 
 To enhance the efficiency of job handling, the `QueueManager` uses a job buffer. This buffer is a temporary storage area for jobs before they are processed, minimizing database access and enabling the system to handle bursts of jobs more effectively.
 
-### Why Add a Signal Handler?
+### Why use a Signal Handler?
 
-Incorporating a signal handler that sets a flag (such as `alive` to `False`) is essential for ensuring a graceful shutdown of the `QueueManager`. When the application receives a termination signal (e.g., SIGINT or SIGTERM), the signal handler can set the `alive` flag to `False`, informing the `QueueManager` to stop processing new jobs and complete the jobs currently in the buffer. This helps maintain the integrity and consistency of the job queue by ensuring that no jobs are left unprocessed or partially processed.
+Using an asyncio.Event for the shutdown event ensures that the shutdown process integrates smoothly with the asynchronous nature of QueueManager. When the application receives a termination signal (e.g., SIGINT or SIGTERM), the signal handler sets the shutdown event, which allows the main loop to detect this and stop processing new jobs while completing the current ones.
 
-In this setup, the `handle_signal` signal handler sets `alive` to `False` upon receiving a termination signal. The main loop checks the `alive` flag, and if set to `False`, it stops running the `QueueManager`, ensuring a clean and graceful shutdown.
+This mechanism helps maintain the integrity and consistency of the job queue, ensuring that all jobs in progress are finalized properly before the system shuts down, thereby preventing data corruption or incomplete tasks.
 
 ## Custom Job Executors
 

--- a/pgqueuer/buffers.py
+++ b/pgqueuer/buffers.py
@@ -43,7 +43,7 @@ class TimedOverflowBuffer(Generic[T]):
             regardless of size.
         flush_callable (Callable[[list[T]], Awaitable[None]]): The asynchronous
             callable used to flush items, such as logging to a database.
-        alive (asyncio.Event): An event to signal when the buffer should stop monitoring
+        shutdown (asyncio.Event): An event to signal when the buffer should stop monitoring
             (e.g., during shutdown).
         events (asyncio.Queue[T]): An asynchronous queue holding the buffered items.
         lock (asyncio.Lock): A lock to ensure thread safety during flush operations.
@@ -54,7 +54,7 @@ class TimedOverflowBuffer(Generic[T]):
     timeout: timedelta
     flush_callable: Callable[[list[T]], Awaitable[None]]
 
-    alive: asyncio.Event = dataclasses.field(
+    shutdown: asyncio.Event = dataclasses.field(
         init=False,
         default_factory=asyncio.Event,
     )

--- a/pgqueuer/db.py
+++ b/pgqueuer/db.py
@@ -111,7 +111,7 @@ class Driver(Protocol):
         raise NotImplementedError
 
     @property
-    def alive(self) -> asyncio.Event:
+    def shutdown(self) -> asyncio.Event:
         """
         An asyncio.Event indicating the liveness of the driver.
 
@@ -161,7 +161,7 @@ class AsyncpgDriver(Driver):
         connection: asyncpg.Connection,
     ) -> None:
         """Initialize the driver with an AsyncPG connection."""
-        self._alive = asyncio.Event()
+        self._shutdown = asyncio.Event()
         self._connection = connection
         self._lock = asyncio.Lock()
 
@@ -193,8 +193,8 @@ class AsyncpgDriver(Driver):
             )
 
     @property
-    def alive(self) -> asyncio.Event:
-        return self._alive
+    def shutdown(self) -> asyncio.Event:
+        return self._shutdown
 
     @property
     def tm(self) -> tm.TaskManager:
@@ -222,7 +222,7 @@ class AsyncpgPoolDriver(Driver):
         """
         Initialize the AsyncpgPoolDriver with a connection pool.
         """
-        self._alive = asyncio.Event()
+        self._shutdown = asyncio.Event()
         self._pool = pool
         self._listener_connection: asyncpg.pool.PoolConnectionProxy | None = None
         self._lock = asyncio.Lock()
@@ -261,8 +261,8 @@ class AsyncpgPoolDriver(Driver):
             )
 
     @property
-    def alive(self) -> asyncio.Event:
-        return self._alive
+    def shutdown(self) -> asyncio.Event:
+        return self._shutdown
 
     @property
     def tm(self) -> tm.TaskManager:
@@ -318,7 +318,7 @@ class PsycopgDriver(Driver):
         notify_timeout: timedelta = timedelta(seconds=0.25),
         notify_stop_after: int = 10,
     ) -> None:
-        self._alive = asyncio.Event()
+        self._shutdown = asyncio.Event()
         self._connection = connection
         self._lock = asyncio.Lock()
         self._tm = tm.TaskManager()
@@ -326,8 +326,8 @@ class PsycopgDriver(Driver):
         self._notify_timeout = notify_timeout
 
     @property
-    def alive(self) -> asyncio.Event:
-        return self._alive
+    def shutdown(self) -> asyncio.Event:
+        return self._shutdown
 
     @property
     def tm(self) -> tm.TaskManager:
@@ -369,13 +369,13 @@ class PsycopgDriver(Driver):
             await self._connection.execute(f"LISTEN {channel};")
 
             async def notify_handler() -> None:
-                while not self.alive.is_set():
+                while not self.shutdown.is_set():
                     gen = self._connection.notifies(
                         timeout=self._notify_timeout.total_seconds(),
                         stop_after=self._notify_stop_after,
                     )
                     async for note in gen:
-                        if not self.alive.is_set():
+                        if not self.shutdown.is_set():
                             callback(note.payload)
                     await asyncio.sleep(self._notify_timeout.total_seconds())
 
@@ -390,5 +390,5 @@ class PsycopgDriver(Driver):
         return self
 
     async def __aexit__(self, *_: object) -> None:
-        self.alive.set()
+        self.shutdown.set()
         await self.tm.gather_tasks()

--- a/pgqueuer/heartbeat.py
+++ b/pgqueuer/heartbeat.py
@@ -20,7 +20,7 @@ class Heartbeat:
     job_id: models.JobId
     interval: timedelta
     buffer: buffers.HeartbeatBuffer
-    alive: asyncio.Event = field(
+    shutdown: asyncio.Event = field(
         init=False,
         default_factory=asyncio.Event,
     )
@@ -38,7 +38,7 @@ class Heartbeat:
         Returns:
             Heartbeat: The Heartbeat instance itself.
         """
-        self.alive.set()  # Activate heartbeats
+        self.shutdown.set()  # Activate heartbeats
         if self.interval > timedelta(seconds=0):
             self.schedule_heartbeat()
         return self
@@ -54,7 +54,7 @@ class Heartbeat:
             exc_val: The exception value, if any.
             exc_tb: The traceback, if any.
         """
-        self.alive.clear()  # Deactivate heartbeats
+        self.shutdown.clear()  # Deactivate heartbeats
         if self.handle:
             self.handle.cancel()
         await self.buffer.flush()
@@ -73,7 +73,7 @@ class Heartbeat:
         """
         Send a heartbeat by adding a JobId to the buffer and scheduling the next heartbeat.
         """
-        if not self.alive.is_set():
+        if not self.shutdown.is_set():
             return
 
         try:
@@ -81,5 +81,5 @@ class Heartbeat:
         except Exception as e:
             logconfig.logger.exception("Failed to send heartbeat: %s", e)
         finally:
-            if self.alive.is_set():
+            if self.shutdown.is_set():
                 self.schedule_heartbeat()

--- a/pgqueuer/qm.py
+++ b/pgqueuer/qm.py
@@ -38,7 +38,7 @@ class QueueManager:
     Attributes:
         connection (db.Driver): The database driver used for database operations.
         channel (models.PGChannel): The PostgreSQL channel for notifications.
-        alive (asyncio.Event): Event to signal when the QueueManager is shutting down.
+        shutdown (asyncio.Event): Event to signal when the QueueManager is shutting down.
         queries (queries.Queries): Instance for executing database queries.
         entrypoint_registry (dict[str, JobExecutor]): Registered job executors.
         entrypoint_statistics (dict[str, models.EntrypointStatistics]): Statistics for entrypoints.
@@ -51,7 +51,7 @@ class QueueManager:
         default=models.PGChannel(queries.DBSettings().channel),
     )
 
-    alive: asyncio.Event = dataclasses.field(
+    shutdown: asyncio.Event = dataclasses.field(
         init=False,
         default_factory=asyncio.Event,
     )
@@ -72,6 +72,11 @@ class QueueManager:
         init=False,
         default_factory=dict,
     )
+
+    @property
+    def alive(self) -> asyncio.Event:
+        # For backwards compatibility
+        return self.shutdown
 
     def __post_init__(self) -> None:
         """
@@ -294,9 +299,9 @@ class QueueManager:
                 self.job_context,
             )
 
-            alive_task = asyncio.create_task(self.alive.wait())
+            shutdown_task = asyncio.create_task(self.shutdown.wait())
 
-            while not self.alive.is_set():
+            while not self.shutdown.is_set():
                 entrypoints = {
                     x: (
                         self.entrypoint_registry[x].retry_timer,
@@ -349,15 +354,15 @@ class QueueManager:
                     dequeue_timeout,
                 )
                 await asyncio.wait(
-                    (alive_task, event_task),
+                    (shutdown_task, event_task),
                     return_when=asyncio.FIRST_COMPLETED,
                 )
 
         if not event_task.done():
             event_task.cancel()
 
-        if not alive_task.done():
-            alive_task.cancel()
+        if not shutdown_task.done():
+            shutdown_task.cancel()
 
     async def _dispatch(
         self,

--- a/pgqueuer/supervisor.py
+++ b/pgqueuer/supervisor.py
@@ -60,7 +60,7 @@ async def runit(
         Handle incoming signals to perform a graceful shutdown.
 
         When a termination signal is received (e.g., SIGINT or SIGTERM), this function
-        sets the 'alive' event in the QueueManager to initiate a controlled shutdown
+        sets the 'shutdown' event in the QueueManager to initiate a controlled shutdown
         process, allowing tasks to complete cleanly.
 
         Args:
@@ -68,7 +68,7 @@ async def runit(
             frame (object): The current stack frame (unused).
         """
         print(f"Received signal {signum}, shutting down...")
-        qm.alive.set()
+        qm.shutdown.set()
 
     # Setup signal handlers for graceful shutdown of the QueueManager.
     signal.signal(signal.SIGINT, graceful_shutdown)  # Handle Ctrl-C

--- a/test/test_cancellation.py
+++ b/test/test_cancellation.py
@@ -39,7 +39,7 @@ async def test_cancellation_async(
         await q.mark_job_as_cancelled(job_ids)
         event.set()
 
-        qm.alive.set()
+        qm.shutdown.set()
 
     await asyncio.gather(
         qm.run(dequeue_timeout=timedelta(seconds=0.0)),
@@ -82,7 +82,7 @@ async def test_cancellation_sync(
         await q.mark_job_as_cancelled(job_ids)
         event.set()
 
-        qm.alive.set()
+        qm.shutdown.set()
 
     await asyncio.gather(
         qm.run(dequeue_timeout=timedelta(seconds=0.01)),
@@ -124,7 +124,7 @@ async def test_cancellation_async_context_manager(
         await q.mark_job_as_cancelled(job_ids)
         event.set()
 
-        qm.alive.set()
+        qm.shutdown.set()
 
     await asyncio.gather(
         qm.run(dequeue_timeout=timedelta(seconds=0.01)),

--- a/test/test_concurrency_limit.py
+++ b/test/test_concurrency_limit.py
@@ -73,7 +73,7 @@ async def test_max_concurrency(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     await asyncio.gather(timer(), dequeue())
 

--- a/test/test_executors.py
+++ b/test/test_executors.py
@@ -158,7 +158,7 @@ async def test_queue_manager_with_custom_executor(apgdriver: Driver) -> None:
 
     task = asyncio.create_task(run_queue_manager())
     await asyncio.sleep(1)  # Allow some time for the job to be processed
-    qm.alive.set()  # Signal the queue manager to stop
+    qm.shutdown.set()  # Signal the queue manager to stop
     await task
 
     assert results == [b"test_data"]

--- a/test/test_qm.py
+++ b/test/test_qm.py
@@ -18,7 +18,7 @@ async def wait_until_empty_queue(
         await asyncio.sleep(0.01)
 
     for qm in qms:
-        qm.alive.set()
+        qm.shutdown.set()
 
 
 @pytest.mark.parametrize("N", (1, 2, 32))
@@ -130,7 +130,7 @@ async def test_pick_local_entrypoints(
     async def waiter() -> None:
         while sum(x.count for x in await q.queue_size() if x.entrypoint == "to_be_picked"):
             await asyncio.sleep(0.01)
-        qm.alive.set()
+        qm.shutdown.set()
 
     await asyncio.gather(
         qm.run(dequeue_timeout=timedelta(seconds=0.01)),

--- a/test/test_retry.py
+++ b/test/test_retry.py
@@ -32,7 +32,7 @@ async def test_retry(
     async def until_retry_updated() -> None:
         while len(seen) <= N:
             await asyncio.sleep(0)
-        c.alive.set()
+        c.shutdown.set()
         e.set()
 
     async with async_timeout.timeout(retry_timer.total_seconds() * 2 * N):
@@ -67,7 +67,7 @@ async def test_no_retry_on_zero_timer(
 
     async def until_retry_updated() -> None:
         await asyncio.sleep(1)
-        c.alive.set()
+        c.shutdown.set()
         e.set()
 
     await asyncio.gather(
@@ -98,7 +98,7 @@ async def test_heartbeat_updates(
     async def until_retry_updated() -> None:
         while len(seen) < N:
             await asyncio.sleep(0)
-        c.alive.set()
+        c.shutdown.set()
         e.set()
 
     async with async_timeout.timeout(retry_timer.total_seconds() * 2 * N):
@@ -138,7 +138,7 @@ async def test_concurrent_retry_jobs(
     async def until_retry_updated() -> None:
         while len(seen_job1) < N or len(seen_job2) < N:
             await asyncio.sleep(0)
-        c.alive.set()
+        c.shutdown.set()
         e1.set()
         e2.set()
 
@@ -192,7 +192,7 @@ async def test_varying_retry_timers(
     async def until_retry_updated() -> None:
         while len(seen_short) < N or len(seen_long) < N:
             await asyncio.sleep(0)
-        c.alive.set()
+        c.shutdown.set()
         e_short.set()
         e_long.set()
 
@@ -235,7 +235,7 @@ async def test_retry_with_cancellation(
     async def until_retry_updated() -> None:
         while len(seen) < N:
             await asyncio.sleep(0)
-        c.alive.set()
+        c.shutdown.set()
         e.set()
 
     async with async_timeout.timeout(retry_timer.total_seconds() * 2 * N):

--- a/test/test_rps.py
+++ b/test/test_rps.py
@@ -65,7 +65,7 @@ async def test_rps(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     await asyncio.gather(timer(), dequeue())
     assert 100 <= tally.count / wait <= 140

--- a/test/test_strict_serialized.py
+++ b/test/test_strict_serialized.py
@@ -89,7 +89,7 @@ async def test_serialized_dispatch(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     await asyncio.gather(timer(), dequeue())
 
@@ -112,7 +112,7 @@ async def test_serialized_dispatch_mixed(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     _, consumed = await asyncio.gather(timer(), dequeue())
     serialized_dispatch_true = list(
@@ -151,7 +151,7 @@ async def test_no_jobs_processed_when_locked(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     _, consumed = await asyncio.gather(timer(), dequeue())
     serialized_dispatch_true = list(
@@ -180,7 +180,7 @@ async def test_mixed_serialized_and_concurrent_processing(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     _, consumed = await asyncio.gather(timer(), dequeue())
     serialized_dispatch_true = list(
@@ -212,7 +212,7 @@ async def test_single_consumer_serialized_behavior(
     async def timer() -> None:
         await asyncio.sleep(wait)
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
     _, consumed = await asyncio.gather(timer(), dequeue())
     serialized_dispatch_true = list(

--- a/tools/benchmark.py
+++ b/tools/benchmark.py
@@ -62,14 +62,14 @@ async def consumer(
 
 
 async def producer(
-    alive: asyncio.Event,
+    shutdown: asyncio.Event,
     queries: Queries,
     batch_size: int,
     cnt: count,
 ) -> None:
     assert batch_size > 0
     entrypoints = ["syncfetch", "asyncfetch"] * batch_size
-    while not alive.is_set():
+    while not shutdown.is_set():
         await queries.enqueue(
             random.sample(entrypoints, k=batch_size),
             [f"{next(cnt)}".encode() for _ in range(batch_size)],
@@ -161,15 +161,15 @@ Enqueue Batch Size:     {args.enqueue_batch_size}
     await (await querier(args.driver, dsn())).clear_log()
     await (await querier(args.driver, dsn())).clear_queue()
 
-    alive = asyncio.Event()
+    shutdown = asyncio.Event()
     qms = list[QueueManager]()
     tqdm_format_dict = {}
 
-    async def enqueue(alive: asyncio.Event) -> None:
+    async def enqueue(shutdown: asyncio.Event) -> None:
         cnt = count()
         producers = [
             producer(
-                alive,
+                shutdown,
                 await querier(args.driver, dsn()),
                 int(args.enqueue_batch_size),
                 cnt,
@@ -202,38 +202,38 @@ Enqueue Batch Size:     {args.enqueue_batch_size}
             await asyncio.gather(*consumers)
             tqdm_format_dict.update(bar.format_dict)
 
-    async def dequeue_alive_timer(
+    async def dequeue_shutdown_timer(
         qms: list[QueueManager],
-        alive: asyncio.Event,
+        shutdown: asyncio.Event,
     ) -> None:
         _, pending = await asyncio.wait(
             (
                 asyncio.create_task(asyncio.sleep(args.timer.total_seconds())),
-                asyncio.create_task(alive.wait()),
+                asyncio.create_task(shutdown.wait()),
             ),
             return_when=asyncio.FIRST_COMPLETED,
         )
         # Stop producers
-        alive.set()
+        shutdown.set()
         # Stop consumers
         for q in qms:
-            q.alive.set()
+            q.shutdown.set()
 
         for p in pending:
             p.cancel()
 
     def graceful_shutdown(signum: int, frame: object) -> None:
-        alive.set()
+        shutdown.set()
         for qm in qms:
-            qm.alive.set()
+            qm.shutdown.set()
 
     signal.signal(signal.SIGINT, graceful_shutdown)  # Handle Ctrl-C
     signal.signal(signal.SIGTERM, graceful_shutdown)  # Handle termination request
 
     await asyncio.gather(
         dequeue(qms),
-        enqueue(alive),
-        dequeue_alive_timer(qms, alive),
+        enqueue(shutdown),
+        dequeue_shutdown_timer(qms, shutdown),
     )
 
     qsize = await (await querier(args.driver, dsn())).queue_size()

